infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -290,23 +304,104 @@ public void reboot() {
+
+                            for (WriteOperation op : overwrite) {
+                                if (op.getMsg().getPayload().getWriteMode() != WriteMode.REPLEX_STREAM) {
+                                    op.getR().sendResponse(op.getCtx(), op.getMsg(),
+                                            CorfuMsgType.ERROR_OVERWRITE.msg());
+                                } else {
+                                    op.getR().sendResponse(op.getCtx(), op.getMsg(),
+                                            CorfuMsgType.ERROR_REPLEX_OVERWRITE.msg());
+                                }
+                            }
+
+                            overwrite.clear();
+                            processed = 0;
+                        }
+                    }
+
+                    if (currOp == WriteOperation.SHUTDOWN) {
+                        log.info("Shutting down the write processor");
+                        break;
+                    }
+                    if (currOp == null) {
+                        lastOp = null;
+                        continue;
+                    }
+
+                    try {
+                        if (currOp.getMsg().getPayload().getWriteMode() != WriteMode.REPLEX_STREAM) {
+                            streamLog.append(new LogAddress(currOp.getMsg().getPayload().getGlobalAddress(), null),
+                                    currOp.getMsg().getPayload().getData());
+                            writeAck.add(currOp);
+                            dataCache.put(new LogAddress(currOp.getMsg().getPayload().getGlobalAddress(), null),
+                                    currOp.getMsg().getPayload().getData());
+                        } else {
+                            for (UUID streamID : currOp.getMsg().getPayload().getStreamAddresses().keySet()) {
+                                streamLog.append(new LogAddress(currOp.getMsg()
+                                                .getPayload()
+                                                .getStreamAddresses()
+                                                .get(streamID), streamID),
+                                        currOp.getMsg().getPayload().getData());
+                                dataCache.put(new LogAddress(currOp.getMsg()
+                                                .getPayload()
+                                                .getStreamAddresses()
+                                                .get(streamID), streamID),
+                                        currOp.getMsg().getPayload().getData());
+                            }
+                            writeAck.add(currOp);
+
+                        }
+                    } catch (OverwriteException e) {
+                        log.info("Overwrite Exception", e);
+                        overwrite.add(currOp);
+                    }
+
+                    processed++;
+                    lastOp = currOp;
+                }
+            } catch (Exception e) {
```

Comment:
The catch statement is outside of the while-loop, so the thread will exit after logging the exception.